packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -136,12 +136,28 @@ export function createInsightsMiddleware<
       // Otherwise, the `init` call might override it with anonymous user token.
       userTokenBeforeInit = userToken;
     });
-    insightsClient('init', {
-      appId,
-      apiKey,
-      useCookie: true,
-      ...insightsInitParams,
-    });
+
+    const version = insightsClient.version;
+    let canInit = !$$internal; // User provided insights middleware.
+
+    // We only check the `version` for the default insights middleware.
+    if ($$internal && typeof version !== 'undefined') {
```

Comment:
maybe this means now that if it's internal, we just _don't_ init, as the version which will allow partial init is the same one that doesn't need init anymore? We'll need to update tests of course, as for now this is still an opt-out branch

Reply:
Yes it makes sense to me, `init({ partial: true })` should be the same as no `init` now.

Reply:
as well as the useCookie still defaulting to true, it makes sense to keep this branch as-is now, but later we'll be able to remove likely